File: 0006-zigzag-conversion/0006-zigzag-conversion.go
func convert(s string, n int) string {
  if n == 1 || len([]rune(s)) < n {
    return s
  }
  rows := make([]string, n)
  row, step := 0, 1
  n--
  for _, rn := range s {
    rows[row] += string(rn)
    row += step
    if row == n || row == 0 {
      step = -step;
    }
  }
  return strings.Join(rows, "")
}

File: 0006-zigzag-conversion/0006-zigzag-conversion.py
class Solution:
  def convert(self, s: str, n: int) -> str:
    if n == 1 or len(s) < n:
      return s
    rows = [''] * n
    row, step = 0, 1
    n -= 1
    for ch in s:
      rows[row] += ch
      row += step
      if row == n or not row:
        step = -step
    return ''.join(rows)